$OpenBSD: patch-lib_byteorder_h,v 1.1 2021/04/24 11:11:47 ajacoutot Exp $

./lib/byteorder.h:64:28: error: function-like macro '__GNUC_PREREQ' is not defined

Index: lib/byteorder.h
--- lib/byteorder.h.orig
+++ lib/byteorder.h
@@ -61,7 +61,12 @@
 #  define ntohll(x) (x)
 
 #else /* small-endian machines */
-#  if defined(__GNUC__) && __GNUC_PREREQ(4, 3)
+#  if defined(__GNUC__) && !defined(__clang__)
+#    if __GNUC_PREREQ(4, 3)
+#      define GNUC_PREREQ_4_3 1
+#    endif
+#  endif
+#  if defined(GNUC_PREREQ_4_3)
      /* Remove existing macros if present */
 #    undef ntohl
 #    undef htonl
